CMakeLists.txt

```diff
@@ -77,6 +77,10 @@ if(ENABLE_WERROR)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
 endif()
 
+if(CMAKE_CXX_COMPILER MATCHES "dpcpp")
+  message(WARNING "Using the intel dpcpp compiler is not supported") 
```

Comment:
Should we not better throw a `FATAL_ERROR` as we know it will not compile even in serial without a bumped version of `Eigen`?

Reply:
My thinking was that somebody could have the some cutting-edge version of eigen and things could just work. I mainly wanted to express that we don’t support this compiler right now.

Reply:
So even with a new Eigen version, we get it at best in serial as of now. And whoever has `dpcpp` also has `icpx` from OneAPI, so they should just use that one.

Reply:
We could actually tell that in the message.

Reply:
Done